SU-SWS/ansible-playbooks
# SITES-916: Put non-vhost group/dept sites in maintenance mode for final cutover

# READY FOR REVIEW

# Summary
- Allows us to set sites in maintenance mode on the Sites 1.x infrastructure at the start of the migration
- Allows us to take sites out of maintenance mode on the ACSF infrastructure after migration
- Simplifies logic for drush alias specification

# Needed By (Date)
- ASAP

# Criticality
- How critical is this PR on a 1-10 scale? 9/10
- It's required for the sites.stanford.edu cutover

# Steps to Test

1. Use this inventory:
```
[sws-internal]
2015drupalcamp
2016drupalcamp

[products:children]
sws-internal

[sws-internal:vars]
group_ids=206
```
2. Add the following to `migration_vars.yml`:
```
enable_maintenance_mode_source: "TRUE"
disable_maintenance_mode_dest: "TRUE"
```
3. Check out this branch
4. Migrate to `test` using `full-migration-playbook.yml`
5. Verify that maintenance mode is _disabled_ on https://2015drupalcamp-test.sites.stanford.edu and https://2016drupalcamp-test.sites.stanford.edu 
6. Verify that maintenance mode is _enabled_ on https://sites.stanford.edu/2015drupalcamp and https://sites.stanford.edu/2016drupalcamp
7. Set `maintenance_mode=0` on `ds_2015drupalcamp` and `ds_2016drupalcamp` on sites1/sites2

# Affected Projects or Products
- sites.stanford.edu cutover

# Associated Issues and/or People
- JIRA ticket: https://stanfordits.atlassian.net/browse/SITES-916

# See Also
- [PR Checklist](https://gist.github.com/sherakama/0ba17601381e3adbe0cad566ad4d80a5)
